include/ncengine/physics/RigidBody.h

```diff
@@ -1,53 +1,124 @@
+/**
+ * @file RigidBody.h
+ * @copyright Jaremie Romer and McCallister Romer 2024
+ */
 #pragma once
 
 #include "ncengine/ecs/Component.h"
+#include "ncengine/ecs/Transform.h"
 
 #include "DirectXMath.h"
 
 namespace nc::physics
 {
+struct ComponentContext;
+class NcPhysicsImpl2;
+class RigidBody;
+
+/** @brief Handle to internal RigidBody state. */
 using BodyHandle = void*;
-using BodyInterface = void*;
 
+/** @brief Flags for configuring RigidBody behavior. */
+struct RigidBodyFlags
```

Comment:
I don't love this data structure:
Flags shouldn't be used for options that are mutually exclusive (`None` and `ScaleWithTransform`). 

As it stands a user could happily do something like:
`constexpr auto g_flags = nc::physics::RigidBodyFlags::ScaleWithTransform | nc::physics::RigidBodyFlags::None;` 

We can easily prevent by extracting the idea of scaling with transform to a boolean.